Ops/Expr/GenXML.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace PSI;
+
+using System.Xml.Linq;
+
+// An basic XML code generator, implemented using the Visitor pattern
+public class ExprXMLGen : Visitor<XElement> {
+
+   public override XElement Visit (NLiteral literal)
+      => NewNode ("Literal", ("Value", literal.Value.Text)
+                           , ("Type", literal.Type));
+
+   public override XElement Visit (NIdentifier identifier)
+      => NewNode ("Ident", ("Name", identifier.Name.Text)
+                                  , ("Type", identifier.Type));
+
+   public override XElement Visit (NUnary unary) {
+      var a = unary.Expr.Accept (this);
+      var elem =  NewNode ("Unary", ("Op", unary.Op.Kind)
+                                  , ("Type", unary.Type));
+      elem.Add (a);
+      return elem;
+   }
+
+   public override XElement Visit (NBinary binary) {
+      var a= binary.Left.Accept (this); var b = binary.Right.Accept (this);
+      var elem = NewNode ("Binary", ("Op", binary.Op.Kind)
+                                  , ("Type", binary.Type));
+      elem.Add (a); elem.Add (b);
+      return elem;
+   }
+
+   XElement NewNode (string name, params (string, object)[] attribs)
+       => new (name, attribs.Select (a => new XAttribute (a.Item1, a.Item2)));
+}
```

Comment:
Here's a more generalized New routine that allows you to pass both attributes (as string,object) tuples or child nodes (as XElement). With this, you should be able to construct the entire node in a single expression. Note that unlike the other visitors, you don't necessarily need to visit the child nodes first for Unary and Binary - you could visit them inline during the New call. Use this to simplify further. 

```
XElement New (string name, params object[] data) {
   var node = new XElement (name);
   foreach (var d in data) {
      if (d is (string K, object V)) node.SetAttributeValue (K, V);
      else if (d is XElement x) node.Add (d);
      else throw new NotImplementedException ();
   }
}
```

Reply:
Done